mpqp/core/instruction/measurement/basis.py

```diff
@@ -195,10 +221,12 @@ def set_size(self, nb_qubits: int):
         Args:
             nb_qubits: number of qubits in the basis
         """
-        pass
+        Bn = reduce(np.kron, [self.basis_vectors] * nb_qubits, np.eye(1))
```

Comment:
chipotage: maybe add a condition that, when the number of qubits is already matching the size of basis vector, we don't need to perform all the steps below (typically when we call `set_size` in the __init__)